tensorflow/compiler/xla/service/gpu/partition_assignment.cc

```diff
@@ -40,8 +40,13 @@ std::ostream& operator<<(std::ostream& out,
 }
 
 int64 ThreadsPerBlockLimit(const se::DeviceDescription& device_desc) {
-  int64 threads_per_block = device_desc.threads_per_block_limit();
-  if (threads_per_block == 0) {
+  auto threads_per_core = device_desc.threads_per_core_limit();
+  auto blocks_per_core = device_desc.blocks_per_core_limit();
+  int64 threads_per_block;
+  if (threads_per_core != 0 && blocks_per_core != 0) {
```

Comment:
This was broken before too, but we seem to use -1 to mean uninitialized?

Reply:
Gotcha; any preference between 
```suggestion
  if (threads_per_core > 0 && blocks_per_core > 0) {
```
and 
```suggestion
  if (threads_per_core != -1 && blocks_per_core != -1) {
```
(The current implementation is from before the previous PR removed it)

Reply:
I'd go with `> 0` just on Postel's law considerations.  (Also we don't want to divide by zero.)